sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/InputStreamContent.java

```diff
@@ -77,20 +95,95 @@ public ByteBuffer toByteBuffer() {
 
     @Override
     public Flux<ByteBuffer> toFluxByteBuffer() {
-        return FluxUtil.toFluxByteBuffer(this.content, STREAM_READ_SIZE);
+        return FluxUtil.toFluxByteBuffer(this.content.get(), STREAM_READ_SIZE);
     }
 
     @Override
     public boolean isReplayable() {
-        return false;
+        return isReplayable;
+    }
+
+    @Override
+    public BinaryDataContent toReplayableContent() {
+        if (isReplayable) {
+            return this;
+        } else {
+            InputStream inputStream = this.content.get();
+            if (canMarkReset(inputStream, length)) {
+                return createMarkResetContent(inputStream, length);
+            } else {
+                return readAndBuffer(inputStream, length);
+            }
+        }
+    }
+
+    @Override
+    public Mono<BinaryDataContent> toReplayableContentAsync() {
+        if (isReplayable) {
+            return Mono.just(this);
+        } else {
+            InputStream inputStream = this.content.get();
+            if (canMarkReset(inputStream, length)) {
+                return Mono.fromCallable(() -> createMarkResetContent(inputStream, length));
+            } else {
+                return Mono.just(inputStream)
+                    .publishOn(Schedulers.boundedElastic()) // reading stream can be blocking.
+                    .map(ignore -> readAndBuffer(inputStream, length));
+            }
+        }
+    }
+
+    private static boolean canMarkReset(InputStream inputStream, Long length) {
+        return length != null && length < Integer.MAX_VALUE - 8 && inputStream.markSupported();
```

Comment:
`Integer.MAX_VALUE - 8` is a magic number and should be declared as  `MAX_ARRAY_LENGTH` constant